### ISSUE TITLE
Support Spark in Mutect2

### ISSUE THREAD

**Issue by cxfustc** [broadinstitute/gatk#4325]
Mutect2 in GATK4 is much faster than that in GATK3, but the parameters '-nt/-nct' are discarded. The similar variant caller HaplotypeCaller can be parallelized using Spark. When will Mutect2 support Spark too?

**Comment by davidbenjamin** [comment]
@cxfustc We have no plans for now to Sparkify Mutect2 because it is trivially parallelizable.  In our wdl we simply scatter over disjoint intervals and merge the resulting vcfs.  Is there an argument in favor of Spark over this approach?

**Comment by droazen** [comment]
@davidbenjamin It's actually worth exploring, I think -- with a Spark implementation it is much easier for end-users running the GATK directly to run with multiple cores. Multi-process interval-based parallelism is only trivial when using a pipeline runner. 

There is already a beta `HaplotypeCallerSpark` implementation -- it would probably be pretty easy to adapt it to call into a `Mutect2Engine` instead of a `HaplotypeCallerEngine`, I think.

**Comment by cxfustc** [comment]
Thank you for replying!
I have parallelized the GATK4 Mutect2 using thread pools in Java. I tested the parallelized GATK4 Mutect2 using a WGS data with control. The result came out that, about 0.6% variants were different from the original results. I found that the difference was caused by the random number generator in ReservoirDownsampler. The order of the input intervals after parallelism were different from the original, so the random numbers generated for each position with redundant reads were possibly different. Is there any solutions for this problem?
Thank you very much!

**Comment by cxfustc** [comment]
Sorry for replying so late. I did not expect the reply for my question came so soon until I checked my GitHub today.
Yes, I am using the default parameters. I am not sure about the difference after FilterMutectCalls. I will deal with it next week after going back to work.
As for as I know, the default limit for downsampling is 50, so the theoretical maximum coverage is about 5000, assuming the length of reads are 100bp. This may be an abnormal value for NGS WGS datas, but many NGS panel datas have very high coverages and their average coverages on panel are possibly more than 5000. Is there a model for selecting a proper downsampling limit for these high coverage datas?
Thank you very much!
